File: immiglensBE/app/schemas/change_detection.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class PostingSnapshotOut(BaseModel):
    id: int
    job_posting_id: int
    capture_result_id: Optional[int]
    page_hash: Optional[str]
    has_changed: Optional[bool]
    change_summary: Optional[str]
    captured_at: datetime

    model_config = {"from_attributes": True}


class ChangeHistoryItem(BaseModel):
    """Simplified view used for timeline display."""
    snapshot_id: int
    captured_at: datetime
    has_changed: Optional[bool]
    change_summary: Optional[str]
    screenshot_url: Optional[str]  # from linked CaptureResult

File: immiglensFE/src/types/change_detection.ts
export interface PostingSnapshot {
  id: number
  job_url_id: number
  capture_result_id: number | null
  page_hash: string | null
  has_changed: boolean | null
  change_summary: string | null
  captured_at: string
}

export interface ChangeHistoryItem {
  snapshot_id: number
  captured_at: string
  has_changed: boolean | null
  change_summary: string | null
  screenshot_url: string | null
}
